pkg/proxy/proxy/transparent.go

```diff
@@ -87,31 +87,52 @@ func (m PolicyMode) IsValid() bool {
 
 // TransparentProxyService transparently proxies HTTP and HTTPS traffic.
 type TransparentProxyService struct {
-	Proxy      *goproxy.ProxyHttpServer
-	Ca         *tls.Certificate
-	NetworkLog *netlog.NetworkActivityLog
-	Policy     *policy.Policy
-	Mode       PolicyMode
+	Proxy  *goproxy.ProxyHttpServer
+	Ca     *tls.Certificate
+	Policy *policy.Policy
+	Mode   PolicyMode
 
-	mx *sync.Mutex
+	mx         *sync.Mutex
+	networkLog *netlog.NetworkActivityLog
+}
+
+// TransparentProxyServiceOpts defines the optional parameters for creating a TransparentProxyService.
+type TransparentProxyServiceOpts struct {
+	Policy      *policy.Policy
+	SkipLogging bool
 }
 
 // NewTransparentProxyService creates a new TransparentProxyService.
-func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate, mode PolicyMode, pl *policy.Policy) TransparentProxyService {
+func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate, mode PolicyMode, opts TransparentProxyServiceOpts) TransparentProxyService {
 	m := new(sync.Mutex)
 	if !mode.IsValid() {
 		log.Fatalf("Invalid proxy mode specified: %v", mode)
 	}
-	if mode != DisabledMode && pl == nil {
-		log.Fatalf("Invalid policy: %v", pl)
+	if mode != DisabledMode && opts.Policy == nil {
+		log.Fatalf("Invalid policy: %v", opts.Policy)
+	}
+	networkLog := &netlog.NetworkActivityLog{}
+	if !opts.SkipLogging {
+		networkLog = netlog.CaptureActivityLog(p, m)
 	}
 	return TransparentProxyService{
 		Proxy:      p,
 		Ca:         ca,
-		NetworkLog: netlog.CaptureActivityLog(p, m),
 		Mode:       mode,
-		Policy:     pl,
+		Policy:     opts.Policy,
 		mx:         m,
+		networkLog: networkLog,
+	}
+}
+
+// RotateLog clears the activity log if logging is enabled. Otherwise starts capturing network activity.
+func (t *TransparentProxyService) RotateLog() {
+	t.mx.Lock()
+	defer t.mx.Unlock()
+	if t.networkLog == nil {
```

Comment:
we don't expect this, do we? can we just blindly assign to CaptureActivityLog?

Reply:
Filed https://github.com/google/oss-rebuild/issues/116 and removed this function for now.